ImageBuilderRAM/ImagePipeline_SharedAccount_Sample_AmznLinux.yml

```diff
@@ -0,0 +1,387 @@
+                - !Sub "${EC2IBRole.Arn}"
+                - !Sub arn:aws:iam::${AppAccountId}:root
+            Action:
+              - kms:DescribeKey
+              - kms:Encrypt
+              - kms:Decrypt
+              - kms:ReEncrypt*
+              - kms:GenerateDataKey
+              - kms:GenerateDataKeyWithoutPlaintext
+            Resource: '*'
+          - Sid: Allow attachement of persistent resources
+            Effect: Allow
+            Principal:
+              AWS:
+                - !Sub arn:aws:iam::${AppAccountId}:root
+            Action:
+              - kms:CreateGrant
+              - kms:ListGrants
+              - kms:RevokeGrant
+            Resource: '*'
+            Condition:
+              Bool:
+                kms:GrantIsForAWSResource: true
+
+
+  
+  KMSKeyAlias: 
+    Type: AWS::KMS::Alias
+    Properties:
+      AliasName: alias/ImageEncryptionKMSKey
+      TargetKeyId: !Ref ImageEncryptionKMSKey
+  
+  
+  ImageBuilderSNSTopic:
+    Type: 'AWS::SNS::Topic'
+    Properties:
+      TopicName: !Ref ImageBuilderSNSTopicName
+  
+  ImageBuilderSNSPolicy:
+    Type: AWS::SNS::TopicPolicy
+    DependsOn: ImageBuilderSNSTopic
+    Properties:
+      PolicyDocument:
+        Id: AllowBackups
+        Version: '2012-10-17'
+        Statement:
+        - Sid: default-policy-id
+          Effect: Allow
+          Principal: 
+            AWS : "*"
+          Action: 
+            - SNS:GetTopicAttributes
+            - SNS:SetTopicAttributes
+            - SNS:AddPermission
+            - SNS:RemovePermission
+            - SNS:DeleteTopic
+            - SNS:Subscribe
+            - SNS:ListSubscriptionsByTopic
+            - SNS:Publish
+            - SNS:Receive
+          Resource: !Ref ImageBuilderSNSTopic
+          Condition: 
+            StringEquals:
+              AWS:SourceOwner: !Sub "${AWS::AccountId}"
+        - Sid: allow-backup-id
+          Effect: Allow
+          Principal:
+            Service: imagebuilder.amazonaws.com
+          Action: sns:Publish
+          Resource: !Ref ImageBuilderSNSTopic
+      Topics:
+      - !Ref ImageBuilderSNSTopic
+
+  ImageBuilderSNSSubscription:
+    Type: AWS::SNS::Subscription
+    Properties:
+      Endpoint: !Ref NotificationEmail
+      Protocol: email
+      TopicArn: !Ref ImageBuilderSNSTopic
+
+  EC2IBRole:
+    Type: AWS::IAM::Role
+    Metadata:
+      cfn-lint: { 
+        config: { 
+          ignore_checks: [ EIAMPolicyWildcardResource, EIAMPolicyActionWildcard ],
+          ignore_reasons: [ 
+            EIAMPolicyWildcardResource: "Requires full access to function",
+            EIAMPolicyActionWildcard: "Required in order to properly function"
+          ]
+        }
+      }
+    Properties:
+      RoleName: !Ref IamEC2IBRoleName
+      Path: /
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+              - ec2.amazonaws.com
```

Comment:
Use url suffix pseudo parameter instead of hard coding ec2.amazonaws.com, to ensure compatibility with other AWS partitions which may use a different suffix for EC2.

- !Sub "ec2.${AWS::URLSuffix}"